Improve squashing documentation (#4668)
kubernetes/community: contributors/guide/github-workflow.md

  # e, edit <commit> = use commit, but stop for amending
  # s, squash <commit> = use commit, but meld into previous commit
  # f, fixup <commit> = like "squash", but discard this commit's log message
  
  ...
  
  ```

3. Use a command line text editor to change the word `pick` to `fixup` for the commits you want to squash, then save your changes and continue the rebase: 

  ```
  pick 2ebe926 Original commit
  squash 31f33e9 Address feedback
  pick b0315fe Second unit of work
  
  ...
  
  ```
  
  Output (after saving changes) is similar to: 
  
  ```
  [detached HEAD 61fdded] Second unit of work
   Date: Thu Mar 5 19:01:32 2020 +0100
   2 files changed, 15 insertions(+), 1 deletion(-)
   
   ...
   
  Successfully rebased and updated refs/heads/master.
  ```
4. Force push your changes to your remote branch:

  ```
  git push --force
  ```

For mass automated fixups (e.g. automated doc formatting), use one or more
commits for the changes to tooling and a final commit to apply the fixup en
masse. This makes reviews easier.

### Merging a commit

Once you've received review and approval, your commits are squashed, your PR is ready for merging. 

> Merging happens automatically when a Reviewer or Approver approves the PR. If you haven't squashed your commits, they may ask you to do so before approving a PR. .

Reviewer: ```suggestion
Merging happens automatically after both a  Reviewer and Approver have approved the PR. If you haven't squashed your commits, they may ask you to do so before approving a PR.
```

Reviewers cannot approve a PR -- it requires sign off from the approver, but generally is first pass by a reviewer when they sign off with `/lgtm`
